parsons/box/box.py

```diff
@@ -0,0 +1,200 @@
+        Box class
+    """
+
+    def __init__(self, client_id=None, client_secret=None, access_token=None):
+        client_id = check_env('BOX_CLIENT_ID', client_id)
+        client_secret = check_env('BOX_CLIENT_SECRET', client_secret)
+        access_token = check_env('BOX_ACCESS_TOKEN', access_token)
+
+        oauth = boxsdk.OAuth2(
+            client_id=client_id,
+            client_secret=client_secret,
+            access_token=access_token
+        )
+        self.client = boxsdk.Client(oauth)
+
+    def create_folder(self, folder_name, parent_folder=None) -> str:
+        """Create a Box folder
+
+        `Args`:
+            folder_name: str
+               The name to give to the new folder
+            parent_folder: str
+               Folder id of the parent folder in which to create the new folder. If
+               omitted, the default folder will be used.
+
+        `Returns`:
+            str: The Box id of the newly-created folder
+        """
+        parent_folder = parent_folder or '0'
+        subfolder = self.client.folder(parent_folder).create_subfolder(folder_name)
+        return subfolder.id
+
+    def delete_folder(self, folder_id) -> None:
+        """Delete a Box folder
+
+        `Args`:
+            folder_id:str
+               The Box id of the folder to delete.
+        """
+        self.client.folder(folder_id=folder_id).delete()
+
+    def delete_file(self, file_id) -> None:
+        """Delete a Box file
+
+        `Args`:
+            file_id:str
+              The Box id of the file to delete.
+        """
+        self.client.file(folder_id=file_id).delete()
+
+    def list_files(self, folder_id='0') -> Table:
+        """List all Box files in a folder
+
+        `Args`:
+            folder_id:str
+               The Box id of the folder in which to search. If omitted,
+               search in the default folder.
+        `Returns`:Table
+            A Parsons table of files and their attributes
+        """
+        return self.list_items(folder_id=folder_id, item_type='file')
+
+    def list_folders(self, folder_id='0') -> Table:
+        """List all Box folders
+
+        `Args`:
+            folder_id:str
+               The Box id of the folder in which to search. If omitted,
+               search in the default folder.
+        `Returns`:Table
+            A Parsons table of folders and their attributes
+        """
+        return self.list_items(folder_id=folder_id, item_type='folder')
+
+    def list_items(self, folder_id='0', item_type=None) -> Table:
+        url = 'https://api.box.com/2.0/folders/' + folder_id
+        json_response = self.client.make_request('GET', url)
+
+        items = Table(json_response.json()['item_collection']['entries'])
+        if item_type:
+            items = items.select_rows(lambda row: row.type == item_type)
+        return items
+
+    # In what formats can we upload/save Tables to Box? For now, just csv
+    ALLOWED_FILE_FORMATS = ['csv', 'json']
```

Comment:
Your comment says we can only save as csv but the allowed file formats includes json?

Reply:
Again, good catch - I added JSON in a subsequent CL, and missed some of the doc updates